doc/further-reading/advanced_identity.rst

```diff
@@ -0,0 +1,103 @@
+
+Advanced attestation service usage
+==================================
+
+This document assumes you have a basic understanding of identification flows in IPv8, as documented in `the overlay tutorial <../../basics/identity_tutorial>`_.
+This document addresses the following three topics:
+
+- Enabling anonymization.
+- Setting a REST API access token.
+- Setting a rendezvous token.
+
+Each of these features can be enabled independently, but should all be enabled when dealing with actual identity data.
+
+
+Enabling anonymization
+----------------------
+
+**Purpose:** *disallow device fingerprinting.*
+
+In the basic identity tutorial we created the following configuration:
+
+.. code-block:: python
+
+    for peer_id in [1, 2]:
+        configuration = get_default_configuration()
+        configuration['keys'] = [{'alias': "anonymous id", 'generation': u"curve25519", 'file': f"keyfile_{peer_id}.pem"}]
+        configuration['working_directory'] = f"state_{peer_id}"
+        configuration['overlays'] = []
+
+To enable anonymization of all traffic through the identity layer we need to load the anonymization overlay.
+This is done by editing the loaded overlays through ``configuration['overlays']``, as follows:
+
+.. code-block:: python
+
+    for peer_id in [1, 2]:
+        configuration = get_default_configuration()
+        configuration['keys'] = [{'alias': "anonymous id", 'generation': u"curve25519", 'file': f"keyfile_{peer_id}.pem"}]
+        configuration['working_directory'] = f"state_{peer_id}"
+        configuration['overlays'] = [overlay for overlay in configuration['overlays'] if overlay['class'] == 'HiddenTunnelCommunity']
+
+Inclusion of the ``'HiddenTunnelCommunity'`` overlay automatically enables anonymization of identity traffic.
+Note that this anonymization:
+
+1. Requires other peers to be online.
+2. Requires additional startup time. It can take several seconds for an anonymized circuit to be established.
+
+Setting a REST API key
+----------------------
+
+**Purpose:** *disallow other local services from hijacking IPv8.*
+
+In the basic identity tutorial we started the REST API as follows:
+
+.. code-block:: python
+
+    for peer_id in [1, 2]:
+        ipv8 = IPv8(configuration)
+        await ipv8.start()
+        rest_manager = RESTManager(ipv8)
+        await rest_manager.start(14410 + peer_id)
+
+To set a REST API key, we will have to pass it to the ``RESTManager`` constructor, as follows (replacing ``"my secret key"`` with your key):
+
+.. code-block:: python
+
+    for peer_id in [1, 2]:
+        ipv8 = IPv8(configuration)
+        await ipv8.start()
+        rest_manager = RESTManager(ipv8, api_key="my secret key")
+        await rest_manager.start(14410 + peer_id)
+
+All requests to the core will then have to use either:
+
+1. The ``X-Api-Key`` HTTP header set to the key value.
+2. A URL parameter ``apikey`` set to the key value.
```

Comment:
You should probably make a note that 1 is the preferred way of adding the API key to an HTTP request, and 2 is merely there to easy development.

Reply:
Done 👍